src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java

```diff
@@ -292,6 +297,14 @@ private String extractGroupId(String appId) {
 		return groupId;
 	}
 
+	private Collection<Constraint> deduceConstraints(AppDeploymentRequest request) {
+		Set<Constraint> requestSpecific = StringUtils.commaDelimitedListToSet(request.getDeploymentProperties().get("spring.cloud.deployer.marathon.constraints"))
+			.stream().map(Constraint::new).collect(Collectors.toSet());
+		Set<Constraint> result = new HashSet<>(properties.getConstraints());
+		result.addAll(requestSpecific);
+		return result;
+	}
+
 	private Collection<String> deduceUris(AppDeploymentRequest request) {
 		Set<String> additional = StringUtils.commaDelimitedListToSet(request.getDeploymentProperties().get("spring.cloud.deployer.marathon.uris"));
```

Comment:
I see this issue in some other places as well. I think we definitely need a constant for the `prefix` and reuse everywhere the properties are used.